src/index.js

```diff
@@ -38,6 +40,12 @@ function handleExtendsNodes(tree, options, messages) {
       throw getError(errors.EXTENDS_NO_SRC);
     }
 
+    try {
+      let locals = extendsNode.attrs.locals ? JSON.parse(extendsNode.attrs.locals) : {};
+      options.expressions.locals = {...options.expressions.locals, ...locals};
+      options.plugins.push(expressions(options.expressions));
+    } catch (e) {}
```

Comment:
This piece can be made more elegant if you immediately define it `locals` with default value `{}`

```js
if (extendsNode.attrs.locals) {
  let locals = {};
  try {
    locals = JSON.parse(extendsNode.attrs.locals);
  } catch (e) {}
    
  options.expressions.locals = merge(options.expressions.locals, locals});
  options.plugins.push(expressions(options.expressions));
}
```
Note that I used the package to merge the data [merge](https://github.com/TehShrike/deepmerge) which also needs to be added since destructuring does not guarantee the correct depth-first joining of data.

Reply:
Do we really need both the if AND the try statement around the JSON parse? Shouldn't just one be enough?

Reply:
> Do we really need both the if AND the try statement around the JSON parse? Shouldn't just one be enough?

`try` works a little slower than the classic `if`. We do not need to try to run parse algorithms and throw an exception if there is none a priori.